pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -125,9 +124,6 @@ func (c *ClientCache) clientConfigForVersion(requiredVersion *schema.GroupVersio
 	}
 	config.GroupVersion = negotiatedVersion
 
-	// TODO this isn't what we want.  Each clientset should be setting defaults as it sees fit.
-	oldclient.SetKubernetesDefaults(&config)
```

Comment:
I thought they did, but it doesn't look like, I'm still trying to figure out how this should work. Just dropping this makes gazillion of tests to fail :/